File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/entity/AlreadyRead.java
package com.amakakeru.mangaworld.entity;

import com.amakakeru.mangaworld.entity.IdClasses.AlreadyReadId;
import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@IdClass(AlreadyReadId.class)
@Table(name = "already_read")
public class AlreadyRead implements Serializable {

    @Id
    @Column(name = "user_id")
    private Long userId;

    @Id
    @Column(name = "m_id")
    private Long mId;

    @Column(name = "added_date", nullable = false)
    private java.sql.Timestamp addedDate;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;
}

File: Frontend/manga-world/src/Model/AlreadyRead.ts
import Manga from "./Manga";
import User from "./User";

class AlreadyRead {
    userid: number;
    mid: number;
    addeddate: Date;
    user: User;
    manga: Manga;

    constructor(userid: number, mid: number, addeddate: Date, user: User, manga: Manga) {
        this.userid = userid;
        this.mid = mid;
        this.addeddate = addeddate;
        this.user = user;
        this.manga = manga;
    }
}

export default AlreadyRead;